File: app/src/main/java/net/transcode001/creambox/asyncs/GetImageTask.java
package net.transcode001.creambox.asyncs;

import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.os.AsyncTask;
import android.widget.ImageView;

import net.transcode001.creambox.Utils.IconCacheUtils;

import java.io.IOException;
import java.io.InputStream;
import java.net.MalformedURLException;
import java.net.URL;


public class GetImageTask extends AsyncTask<Void,Void,Bitmap> {
    private ImageView imageView;
    private String tag;
    private twitter4j.Status status;
    private IconCacheUtils utils;

    public GetImageTask(ImageView imageView,twitter4j.Status status,IconCacheUtils utils){
        this.imageView = imageView;
        this.status = status;
        tag = imageView.getTag().toString();
        this.utils = utils;
    }

    @Override
    protected Bitmap doInBackground(Void... params){
        Bitmap pic = utils.getIcon(status.getUser().getScreenName());
        if(pic!=null) return pic;

        try{
            URL url = new URL(status.getUser().getMiniProfileImageURL());
            InputStream mStream = url.openStream();
            Bitmap bmp = BitmapFactory.decodeStream(mStream);
            mStream.close();
            return bmp;

        }catch(MalformedURLException me){
            me.printStackTrace();
            return null;
        }catch(IOException ioe){
            ioe.printStackTrace();
            return null;
        }
    }

    @Override
    protected void onPostExecute(Bitmap bmp){
        if(bmp!=null){
            if(tag.equals(this.imageView.getTag())){
                imageView.setImageBitmap(bmp);
                utils.setIcon(status.getUser().getScreenName(),bmp);
            }
        }
    }
}

File: app/src/main/java/net/transcode001/creambox/asyncs/GetImageTask.kt
package net.transcode001.creambox.asyncs

import android.graphics.Bitmap
import android.graphics.BitmapFactory
import android.os.AsyncTask
import android.widget.ImageView

import net.transcode001.creambox.Utils.IconCacheUtils

import java.io.IOException
import java.io.InputStream
import java.net.MalformedURLException
import java.net.URL


internal class GetImageTask(private val imageView: ImageView, private val status: twitter4j.Status, private val utils: IconCacheUtils) : AsyncTask<Void, Void, Bitmap>() {
    private val tag: String

    init {
        tag = imageView.tag.toString()
    }

    override fun doInBackground(vararg params: Void): Bitmap? {
        val pic = utils.getIcon(status.user.screenName)
        if (pic != null) return pic

        try {
            val url = URL(status.user.profileImageURL)
            val mStream = url.openStream()
            val bmp = BitmapFactory.decodeStream(mStream)
            mStream.close()
            return bmp

        } catch (me: MalformedURLException) {
            me.printStackTrace()
            return null
        } catch (ioe: IOException) {
            ioe.printStackTrace()
            return null
        }

    }

    override fun onPostExecute(bmp: Bitmap?) {
        if (bmp != null) {
            if (tag == this.imageView.tag) {
                imageView.setImageBitmap(bmp)
                utils.setIcon(status.user.screenName, bmp)
            }
        }
    }
}
